### PR TITLE
Review request for Microsoft.RecoveryServices to add version stable/2023-01-01

### PR DESCRIPTION
**Pull request by singhabh27**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by raosuhas** [comment]
>         }

For both these definitions please consider using enums insead of the list of booleans

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)

Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].


---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24474 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by singhabh27** [comment]
Hi @raosuhas, this is as per design. We use ASR bool converter in our service code. Existing example
https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535#:~:text=%22-,isOSDisk,-%22%3A%20%7B

**Comment by visingla-ms** [comment]
>     "VMwareCbtSecurityProfileDetails": {

VMwareCbtSecurityProfileDetails and VMwareCbtSecurityProfileInput, both have same schema. Why are these two being defined as two objects? I assume, both Input and Response can use same schema, right?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24416 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by visingla-ms** [comment]
>         "targetVmSecurityType": {

Observed that targetVmSecurityType in this VMwareCbtSecurityProfileDetails object is of type string, whereas in VMwareCbtSecurityProfileInput, it is of type enum?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24420 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by visingla-ms** [comment]
>         }

These 4 properties start with "is" indicating they should be booleans (or better, enums)? Why are these strings?

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24474 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by visingla-ms** [comment]
>           "description": "A value indicating the inplace OS Upgrade version.",

The description doesn't seem to be apt? Do you mean something like: List of supported inplace OS Upgrade versions.

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24047 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by singhabh27** [comment]
> > ```
> > "VMwareCbtSecurityProfileDetails": {
> > ```
> 
> VMwareCbtSecurityProfileDetails and VMwareCbtSecurityProfileInput, both have same schema. Why are these two being defined as two objects? I assume, both Input and Response can use same schema, right?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24416 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)

This is as per design. We use seperate objects for put and get calls.

**Comment by singhabh27** [comment]
> > ```
> >     "targetVmSecurityType": {
> > ```
> 
> Observed that targetVmSecurityType in this VMwareCbtSecurityProfileDetails object is of type string, whereas in VMwareCbtSecurityProfileInput, it is of type enum?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24420 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)

 VMwareCbtSecurityProfileDetails is used in Get call thats why kept as string.

**Comment by singhabh27** [comment]
> We use ASR bool converter in our service code. Example of existing case:
> image

We use ASR bool converter in our service code. Example of existing case:
image

**Comment by singhabh27** [comment]
> > ```
> >     }
> > ```
> 
> These 4 properties start with "is" indicating they should be booleans (or better, enums)? Why are these strings?
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24474 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)

> We use ASR bool converter in our service code. Example of existing case:
> image

**Comment by singhabh27** [comment]
> > ```
> >       "description": "A value indicating the inplace OS Upgrade version.",
> > ```
> 
> The description doesn't seem to be apt? Do you mean something like: List of supported inplace OS Upgrade versions.
> 
> Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24047 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)

Yes, this represents list of supported inplace OS Upgrade versions.

**Comment by visingla-ms** [comment]
> > > ```
> > >     }
> > > ```
> > 
> > 
> > These 4 properties start with "is" indicating they should be booleans (or better, enums)? Why are these strings?
> > Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24474 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)
> 
> > We use ASR bool converter in our service code. Example of existing case:
> > image

Any particular issue you observed with using boolean nonetheless, which will not even need that conversion? That will make your resource model intuitive and lesser processing for payload.

**Comment by visingla-ms** [comment]
> > > ```
> > > "VMwareCbtSecurityProfileDetails": {
> > > ```
> > 
> > 
> > VMwareCbtSecurityProfileDetails and VMwareCbtSecurityProfileInput, both have same schema. Why are these two being defined as two objects? I assume, both Input and Response can use same schema, right?
> > Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24416 in [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535). [](commit_id = [c7d7b97](https://github.com/Azure/azure-rest-api-specs/commit/c7d7b97367a05da96340f025cdd6405c37b23535), deletion_comment = False)
> 
> This is as per design. We use seperate objects for put and get calls.

But there is no difference between the two (except for the targetVmSecurityType being enum vs string which can be enum in both cases without any issues). ARM RPC recommends having Put Payload and Get Response with same schema. It's uniform for clients, and easier to maintain for engineering teams to deal with one resource model.

**Comment by visingla-ms** [comment]
As discussed about making Put request and response payload properties same, please share your thoughts on how are your addressing it.

**Comment by sjanamma** [comment]
>         }

not addressed

---
In reply to: [1400701927](https://github.com/Azure/azure-rest-api-specs/pull/22076#issuecomment-1400701927) [](http://example.com/codeflow?ancestors=1400701927,1400701927)

---
Refers to: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/service.json:24474 in c7d7b97. [](commit_id = c7d7b97367a05da96340f025cdd6405c37b23535, deletion_comment = False)

**Comment by sjanamma** [comment]
PUT Request/Response schema must match with response of the GET calls

---
In reply to: [1401560063](https://github.com/Azure/azure-rest-api-specs/pull/22076#issuecomment-1401560063) [](http://example.com/codeflow?ancestors=1401560063,1401560063)

File: specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/examples/ReplicationFabrics_Purge.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "fabricName": "cloud1",
+    "api-version": "2023-01-01",
+    "resourceName": "vault1",
+    "resourceGroupName": "resourceGroupPS1",
+    "subscriptionId": "c183865e-6077-46f2-a3b1-deb0f4f4650a"
+  },
+  "responses": {
+    "202": {
+      "headers": {
+        "Location": {
```
Comment (dw511214992) [inline specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2023-01-01/examples/ReplicationFabrics_Purge.json:12]: please refer to following example to change the example:
```
    "headers": {
        "location": "https://localhost/subscriptions/651f8027-33e8-4ec4-97b4-f6e9f3dc8744/resourceGroups/workerapps-rg-xj/providers/Microsoft.App/containerApps/myapp/sourcecontrols/currentInOperationModel/operationresults/5a7f31af-8ae5-489b-a67e-f0a2d11df796?api-version=2021-03-01"
      },
```

**Review by lirenhe — APPROVED** [review]

**Pull request merged**